MUXSDKKaltura/MUXSDKKalturaTests/MUXSDKStatsTests.swift

```diff
@@ -64,4 +64,54 @@ class MUXSDKStatsTests: XCTestCase {
 
         XCTAssertEqual(viewerData.viewerApplicationName, "MUX Kaltura Tests Version 2")
     }
+    
+    func testMonitorPlayer() {
+        guard let customerData = MockedData.customerData else {
+            XCTFail("Customer data not found")
+            return
+        }
+        
+        MUXSDKStats.monitorPlayer(player: MockedData.player, playerName: MockedData.playerName, customerData: customerData)
+        let expectedEvents = [
+            MUXSDKPlaybackEventViewInitEventType,
+            MUXSDKDataEventType,
+            MUXSDKPlaybackEventPlayerReadyEventType
+        ]
+        
+        self.assertDispatchedEventTypesForPlayer(
+            id: MockedData.playerName,
+            expectedEventTypes: expectedEvents
+        )
+        
+        let expectedCustomerVideoData: [String : Any] = [
```

Comment:
these expected values are coming directly from `MockedData.customerData` right? Any reason not to reference those values directly like `customerData.videoData.videoTitle`?

Reply:
So I think I started the other tests by hardcoding the expected data that we want to assert in each test, because from past experience when using the assert data from the MockedData it's very easy to run into cases where that structure is updated for a new test, the old test still passes but it loses the original purpose for which it was designed and since the test still passes it is just overlooked. So I prefer to have each test specifying exactly it's own expected data. 

tbh in this case is less important but for example if we have a test that is sensitive to the data like it expects a string with a specific special character and then the mocked data is changed we wouldn't noticed and the test would no longer serve it's original purpose.

Reply:
I do see what you are saying but then it seems to me that any test which is sensitive to the data should be using a mocked object that is not used by other tests? It actually seems like a slippery slope to use the same mocked data from a number of different tests for the exact reason you bring up